src/utils/components/InstanceTypeModal/utils/util.ts

```diff
@@ -80,34 +83,42 @@ const sortInstanceTypeSizes = (a: InstanceTypeRecord, b: InstanceTypeRecord) =>
 
 export const getInstanceTypesPrettyDisplaySize = (
   mappedInstanceTypes: MappedInstanceTypes,
-  instanceTypeSeries: InstanceTypesSeries,
-  instanceTypeSize: InstanceTypeSize,
+  instanceTypeSeries?: InstanceTypesSeries,
+  instanceTypeSize?: InstanceTypeSize,
 ) => mappedInstanceTypes?.[instanceTypeSeries]?.sizes[instanceTypeSize]?.prettyDisplaySize;
 
-export const getInstanceTypesSizes = (mappedInstanceTypes: MappedInstanceTypes, series: string) => {
+export const getInstanceTypesSizes = (
+  mappedInstanceTypes: MappedInstanceTypes,
+  series?: string,
+) => {
   const matchedSeries = Object.values(mappedInstanceTypes).find(
     (it) => it.displayNameSeries === series,
   );
-  return Object.values(matchedSeries?.sizes).sort(sortInstanceTypeSizes);
+  return Object.values(matchedSeries?.sizes ?? {}).sort(sortInstanceTypeSizes);
 };
 
 export const getInstanceTypeSeriesDisplayName = (
   mappedInstanceTypes: MappedInstanceTypes,
-  instanceTypeSeries: InstanceTypesSeries,
+  instanceTypeSeries?: InstanceTypesSeries,
 ) => mappedInstanceTypes?.[instanceTypeSeries]?.displayNameSeries;
 
 export const getInstanceTypeFromSeriesAndSize = (
   mappedInstanceTypes: MappedInstanceTypes,
-  instanceTypeSeries: string,
-  instanceTypeSize: string,
+  instanceTypeSeries?: string,
+  instanceTypeSize?: string,
 ): InstanceTypeUnion => {
   const instanceTypesSeries = Object.values(mappedInstanceTypes);
 
   const matchedSeries = instanceTypesSeries.find(
-    (series) => series.displayNameSeries === instanceTypeSeries,
+    (series) => series?.displayNameSeries === instanceTypeSeries,
   );
-  const matchedSize = Object.values(matchedSeries?.sizes).find(
-    (size) => size.prettyDisplaySize === instanceTypeSize,
+
+  if (!matchedSeries) {
+    return undefined;
```

Comment:
undefined isn't used for return values; undefined in JS usually means uninitialized var. Better return null

Reply:
This is consistent with what we return currently - note that optional chaining operator `?.` and  `find()` return `undefined`.  IMHO having both in one method is overkill.

Reply:
?. make sense to return undefined, https://262.ecma-international.org/6.0/#sec-terms-and-definitions-undefined-type , u can also commit the return value and simply use return , u will get same value